# Generated by Django 3.1.12 on 2022-02-24 17:22

from django.db import migrations, models

from integration import error_codes

bad_gateway_error = {
    'code': error_codes.BAD_GATEWAY,
    'error': 'BAD_GATEWAY_ERROR',
    'default_wizard_message': '',
    'failure_reason_mail': '',
    'send_email': False,
    'description': 'Server received an invalid response from the upstream server',
}


def add_bad_gateway_error_catalogue(apps, _):
    error_catalogue_model = apps.get_model('integration', 'ErrorCatalogue')

    error_catalogue_model.objects.create(**bad_gateway_error)


class Migration(migrations.Migration):
    dependencies = [
        ('integration', '0048_expired_token_code'),
    ]

    operations = [
        migrations.AlterField(
            model_name='connectionaccount',
            name='error_code',
            field=models.CharField(
                choices=[
                    ('000', 'None'),
                    ('001', 'Other'),
                    ('002', 'Insufficient Permissions'),
                    ('003', 'Missing Github Organization'),
                    ('004', 'Denial Of Consent'),
                    ('005', 'Insufficient Config Data'),
                    ('006', 'Invalid Client Credentials'),
                    ('007', 'Expired Token'),
                    ('011', 'Provider Server Error'),
                    ('012', 'Resource not found'),
                    ('013', 'API call exceeded rate limit due to too many requests'),
                    ('014', 'Github App is not installed on the organization'),
                    ('015', 'Provider Graphql API Error'),
                    ('016', 'Connection Time Out'),
                    ('017', 'Default GraphQL Error'),
                    ('018', 'Access Revoked'),
                    ('019', 'The user entered an invalid value'),
                    ('020', 'Gateway Timeout'),
                    ('008', 'Bad Gateway'),
                ],
                default='000',
                max_length=20,
            ),
        ),
        migrations.RunPython(add_bad_gateway_error_catalogue),
    ]
